File: JavaScript/Array/05_Maximize_Stock_Profit.js
class Solution {
    maximizeStockProfit(prices) {
        // prices: list of prices for each day
        // Return the maximum profit one can achieve

         if (prices.length <= 1) return 0
    
    let profit = 0;
    let buy = prices[0];

    for(let i = 1; i < prices.length; i++){
        if(prices[i] < buy){
            buy = prices[i]
        }
        else if( prices[i] - buy > profit){
                profit = prices[i] - buy;
            }
        }
    return profit
    }
};

File: Python/Array/05_Maximize_Stock_Profit.py
class Solution:
    def maximizeStockProfit(self, prices):
        '''
        prices: list of prices for each day
        Return the maximum profit one can achieve
        '''
        profit = 0
        buy = float('inf')

        for i in prices:
            if(i < buy):
                buy = i
            else:
                if(profit < (i - buy)):
                    profit = (i-buy)
        return profit
